src/frontend/screens/ObservationEdit/SaveButton.tsx

```diff
@@ -170,16 +169,18 @@ export const SaveButton = ({
         onSuccess: () => {
           clearDraft();
           navigation.pop();
-          if (value.lat && value.lon) {
-            addNewTrackLocation([
-              {
-                timestamp: new Date().getTime(),
-                latitude: value.lat,
-                longitude: value.lon,
-              },
-            ]);
+          if (isTracking) {
+            addNewTrackObservation(observationId);
+            if (value.lat && value.lon) {
+              addNewTrackLocation([
+                {
+                  timestamp: new Date().getTime(),
+                  latitude: value.lat,
+                  longitude: value.lon,
+                },
+              ]);
+            }
```

Comment:
Sorry I didn't catch this earlier. But we actually don't want to save an observation in a track when they are editting an observation, only creating an observation. If is a user is `tracking` and goes to edit a previous observation, that observation will be unrelated to the track, so we don't want to add it to the track